android/app/src/main/java/com/zulipmobile/MainActivity.kt

```diff
@@ -30,26 +29,23 @@ open class MainActivity : ReactActivity() {
     /* Returns true just if we did handle the intent. */
     private fun maybeHandleIntent(intent: Intent?): Boolean {
         intent ?: return false
-        val url = intent.data
+        val maybeReactContext = reactApplication.tryGetReactContext()
         when (intent.action) {
             // Share-to-Zulip
             Intent.ACTION_SEND, Intent.ACTION_SEND_MULTIPLE -> {
-                handleSend(intent, reactApplication.tryGetReactContext(), contentResolver)
+                handleSend(intent, maybeReactContext, contentResolver)
```

Comment:
I don't think this benefits from getting the React context eagerly; we may as well wait to invoke `reactApplication.tryGetReactContext()` until we know we're going down a codepath that needs it. (IOW, can drop the "Pull out a `maybeReactContext` variable" commit.)